nixos/modules/services/networking/headscale.nix

```diff
@@ -0,0 +1,476 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.headscale;
+
+  dataDir = "/var/lib/headscale";
+  runDir = "/run/headscale";
+
+  settingsFormat = pkgs.formats.yaml { };
+  configFile = settingsFormat.generate "headscale.yaml" cfg.extraSettings;
+in
+{
+  options = {
+    services.headscale = {
+      enable = mkEnableOption "headscale, Open Source coordination server for Tailscale";
+
+      package = mkPackageOption pkgs "headscale";
+
+      user = mkOption {
+        default = "headscale";
+        type = types.str;
+        description = ''
+          User the headscale server should execute under.
+        '';
+      };
+
+      group = mkOption {
+        default = "headscale";
+        type = types.str;
+        description = ''
+          If the default user "headscale" is configured then this is the primary
+          group of that user.
+        '';
+      };
+
+      serverUrl = mkOption {
+        type = types.str;
+        default = "http://127.0.0.1:8080";
+        description = ''
+          The url clients will connect to.
+        '';
+        example = "https://myheadscale.example.com:443";
+      };
+
+      address = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = ''
+          Listening address of headscale.
+        '';
+        example = "0.0.0.0";
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 8080;
+        description = ''
+          Listening port of headscale.
+        '';
+        example = "443";
+      };
+
+      privateKeyFile = mkOption {
+        type = types.path;
+        description = ''
+          Path to private key file, generated automatically if it does not exist.
+        '';
+      };
+
+      derp = {
+        urls = mkOption {
+          type = types.listOf types.str;
+          default = [ "https://controlplane.tailscale.com/derpmap/default" ];
+          description = ''
+            List of urls containing DERP maps.
+
+            <link xlink:href="https://tailscale.com/blog/how-tailscale-works/"><literal>How Tailscale works</link>
```

Comment:
this being xml those will not render as two paragraphs unless explicitly marked as such with `<para>` :(

(really hope the transition to markdown will happen soon enough, docbook is horrid)